hp/src/HpParser.hs

```diff
@@ -30,6 +30,7 @@ import Lexer
 import SymbolTable
 import Text.Parsec hiding (manyAccum)
 import Text.Parsec.String (Parser)
+import ErrorHandler (SemanticError(AlreadyDeclared))
 
```

Comment:
I tested using this example code: [https://github.com/luiggymamani/hufflepuff-own-language/blob/feature/code-generator/hp/code.hp](https://github.com/luiggymamani/hufflepuff-own-language/blob/feature/code-generator/hp/code.hp)

There are parsing errors that do not exist in the main, please correct them:

![Screenshot from 2024-07-04 21-23-19](https://github.com/luiggymamani/hufflepuff-own-language/assets/135136456/b1b234a0-3cd6-45c7-97c3-0aa799618084)

and I don't know if the "Already Declared list" error is also caused by the parser, but the map needs a list or a reference to a list, so it's not creating a new reference